Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1278,57 +1278,37 @@ static HTMLElement* topmostPopoverAncestor(Element& newPopover)
 
     topLayerPositions.add(newPopover, i);
 
+    // https://html.spec.whatwg.org/#nearest-inclusive-open-popover
+    auto nearestInclusiveOpenPopover = [](Element& candidate) -> HTMLElement* {
+        for (auto& element : lineageOfType<HTMLElement>(candidate)) {
+            if (element.popoverState() == PopoverState::Auto && element.popoverData()->visibilityState() == PopoverVisibilityState::Showing)
+                return &element;
+        }
+        return nullptr;
+    };
+
     RefPtr<HTMLElement> topmostAncestor;
 
-    auto checkAncestor = [&](Element* candidate) {
+    auto checkAncestor = [&](HTMLElement* candidate) {
         if (!candidate)
             return;
 
-        // https://html.spec.whatwg.org/#nearest-inclusive-open-popover
-        auto nearestInclusiveOpenPopover = [](Element& candidate) -> HTMLElement* {
-            for (auto& element : lineageOfType<HTMLElement>(candidate)) {
-                if (element.popoverState() == PopoverState::Auto && element.popoverData()->visibilityState() == PopoverVisibilityState::Showing)
-                    return &element;
-            }
-            return nullptr;
-        };
-
         auto* candidateAncestor = nearestInclusiveOpenPopover(*candidate);
         if (!candidateAncestor)
             return;
+
         if (!topmostAncestor || topLayerPositions.get(*topmostAncestor) < topLayerPositions.get(*candidateAncestor))
             topmostAncestor = candidateAncestor;
     };
 
-    checkAncestor(newPopover.parentElementInComposedTree());
+    if (auto* ancestor = newPopover.parentElementInComposedTree())
+        checkAncestor(nearestInclusiveOpenPopover(*ancestor));
```

Comment:
This change (and related) shouldn't be needed, `checkAncestor` calls `nearestInclusiveOpenPopover` anyway. I'll file a spec issue

Reply:
Thanks for filing, removed.